### ISSUE TITLE
Rollback from libarchive/liblzma to seven_zip_ruby to handle self-extracting 7z files

### ISSUE THREAD

**Issue by maxirmx** [fontist/excavate#24]
There is one fontist formula (https://github.com/fontist/formulas/blob/master/Formulas/adobe_reader_20.yml), that uses a  Adobe Acrobat Reader Installation package as font source. 

This package is created with format issue as described here: https://www.mail-archive.com/xz-devel@tukaani.org/msg00370.html
It looks like it is ongoing problem because of toolchain that is used to create this package. 

The bottom line is that we cannot use libarchive/liblzma to handle self-extracting 7z files and have to rollback to seven_zip_ruby

**Comment by maxirmx** [comment]
I suggest we return to seven_zip_ruby, at least for now

excavate was missing tests for this case ( BCJ, LZMA archive ) and for another complex case (CAB file embedded into self-extracting 7z archive) . These are be added

**Comment by ronaldtse** [comment]
Agree, thanks!
